File: src/main/java/com/mbayou/kick4k/events/type/Identity.java
package com.mbayou.kick4k.events.type;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class Identity {
    private final String usernameColor;
    private final List<Badge> badges;

    @JsonCreator
    public Identity(@JsonProperty("username_color") String usernameColor,
                    @JsonProperty("badges") List<Badge> badges) {
        this.usernameColor = usernameColor;
        this.badges = badges;
    }

    public String getUsernameColor() {
        return usernameColor;
    }

    public List<Badge> getBadges() {
        return badges;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/type/Identity.kt
package com.mbayou.kick4k.events.type

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class Identity @JsonCreator constructor(
    @JsonProperty("username_color") val usernameColor: String?,
    @JsonProperty("badges") val badges: List<Badge>?,
)
